### Step 1: Load and Explore Dataset
We begin by loading the fitness dataset into Pandas and exploring its structure.
This helps us understand the features before applying scaling.


In [2]:
import pandas as pd

# Load dataset (replace with actual file name if different)
df = pd.read_csv("fitness_activity_100+_preprocess.csv")

# Explore dataset
print("First 5 rows:\n", df.head())
print("\nShape:", df.shape)
print("\nColumns:", df.columns.tolist())
print("\nData Types:\n", df.dtypes)
print("\nStatistical Summary:\n", df.describe())


First 5 rows:
   User_ID  Steps_Per_Day  Calories_Burned  Workout_Minutes
0      U1           9270              581               91
1      U2          17795              735              119
2      U3           2860              341               17
3      U4           7390              381              117
4      U5          15418              449               84

Shape: (100, 4)

Columns: ['User_ID', 'Steps_Per_Day', 'Calories_Burned', 'Workout_Minutes']

Data Types:
 User_ID            object
Steps_Per_Day       int64
Calories_Burned     int64
Workout_Minutes     int64
dtype: object

Statistical Summary:
        Steps_Per_Day  Calories_Burned  Workout_Minutes
count     100.000000       100.000000       100.000000
mean     9992.390000       542.620000        69.030000
std      4551.436736       209.950681        32.540093
min      2189.000000       184.000000        15.000000
25%      5929.750000       377.250000        40.250000
50%     10107.500000       561.500000        70.5000

### Step 2: Apply Standard Scaling
StandardScaler transforms features so they have mean = 0 and standard deviation = 1.
This is useful when features are measured in different units but we want them on a comparable scale.


In [4]:
from sklearn.preprocessing import StandardScaler

# Select numeric columns
numeric_cols = ['Steps_Per_Day', 'Calories_Burned', 'Workout_Minutes']

# Apply StandardScaler
scaler = StandardScaler()
df_standard = pd.DataFrame(
    scaler.fit_transform(df[numeric_cols]),
    columns=numeric_cols
)

# Keep User_ID alongside scaled data
df_standard = pd.concat([df['User_ID'], df_standard], axis=1)

print("First 5 rows after Standard Scaling:\n", df_standard.head())
print("\nStatistical Summary:\n", df_standard.describe())


First 5 rows after Standard Scaling:
   User_ID  Steps_Per_Day  Calories_Burned  Workout_Minutes
0      U1      -0.159517         0.183726         0.678568
1      U2       1.722954         0.920927         1.543380
2      U3      -1.574958        -0.965159        -1.607006
3      U4      -0.574654        -0.773678         1.481608
4      U5       1.198071        -0.448161         0.462365

Statistical Summary:
        Steps_Per_Day  Calories_Burned  Workout_Minutes
count   1.000000e+02     1.000000e+02     1.000000e+02
mean    1.199041e-16    -3.552714e-17    -3.108624e-17
std     1.005038e+00     1.005038e+00     1.005038e+00
min    -1.723127e+00    -1.716721e+00    -1.668778e+00
25%    -8.971028e-01    -7.916293e-01    -8.889031e-01
50%     2.541833e-02     9.037891e-02     4.540262e-02
75%     7.312069e-01     8.862208e-01     9.642652e-01
max     1.750557e+00     1.705998e+00     1.543380e+00


### Step 3: Apply Min-Max Scaling
Min-Max Scaler rescales features to a fixed range, usually [0,1].
This is useful when we want to preserve relative relationships but normalize values.


In [5]:
from sklearn.preprocessing import MinMaxScaler

# Select numeric columns
numeric_cols = ['Steps_Per_Day', 'Calories_Burned', 'Workout_Minutes']

# Apply Min-Max Scaler
scaler = MinMaxScaler()
df_minmax = pd.DataFrame(
    scaler.fit_transform(df[numeric_cols]),
    columns=numeric_cols
)

# Keep User_ID alongside scaled data
df_minmax = pd.concat([df['User_ID'], df_minmax], axis=1)

print("First 5 rows after Min-Max Scaling:\n", df_minmax.head())
print("\nStatistical Summary:\n", df_minmax.describe())


First 5 rows after Min-Max Scaling:
   User_ID  Steps_Per_Day  Calories_Burned  Workout_Minutes
0      U1       0.450130         0.555245         0.730769
1      U2       0.992054         0.770629         1.000000
2      U3       0.042655         0.219580         0.019231
3      U4       0.330621         0.275524         0.980769
4      U5       0.840951         0.370629         0.663462

Statistical Summary:
        Steps_Per_Day  Calories_Burned  Workout_Minutes
count     100.000000       100.000000       100.000000
mean        0.496052         0.501566         0.519519
std         0.289329         0.293637         0.312886
min         0.000000         0.000000         0.000000
25%         0.237795         0.270280         0.242788
50%         0.503369         0.527972         0.533654
75%         0.706551         0.760490         0.819712
max         1.000000         1.000000         1.000000


### Step 4: Comparative Analysis

**StandardScaler**
- Transforms data so each feature has mean = 0 and standard deviation = 1.
- Produces values that can be negative or greater than 1.
- Best suited for algorithms that assume normally distributed data (e.g., Logistic Regression, PCA, Linear Regression).

**Min-Max Scaler**
- Rescales features to a fixed range [0,1].
- Preserves relative relationships between values.
- Best suited for algorithms sensitive to absolute magnitudes (e.g., Neural Networks, KNN, Gradient Descent-based models).

**Observations from our dataset:**
- StandardScaler centered all features around 0 with unit variance. For example, `Steps_Per_Day` now ranges roughly from -1.7 to +1.7.
- Min-Max Scaler compressed all features into [0,1]. For example, `Steps_Per_Day` now ranges strictly between 0 and 1.
- StandardScaler is useful when features have very different variances and we want them comparable.
- Min-Max Scaler is useful when we want bounded values, especially for distance-based or gradient-based algorithms.

**Conclusion:**
Both scaling methods achieve uniformity but in different ways. Choice depends on the algorithm and the nature of the dataset.
